feat(general): redirection

sshwy opened the pull request:
replace `perform` with `yjudger`

sshwy committed:
chore: change api name

replace `perform` with `yjudger`

ghost commented:
### Chore

* change api name ([011ace7](https://github.com/sshwy/yaoj-judger/commit/011ace70026a36f12844eed3a8ad67e4732df883))
* drop t.in t.out ([0e97f7f](https://github.com/sshwy/yaoj-judger/commit/0e97f7fa468e899e8afabafccb10f211f33594c0))


### Code Refactoring

*  **judger:** interactive ([460b6a4](https://github.com/sshwy/yaoj-judger/commit/460b6a46ebe36779c2a7e9603854692dc7c9b4f1))
* flush log ([d23b41c](https://github.com/sshwy/yaoj-judger/commit/d23b41c174784f7feed25c61b98e420a73f7d3cf))
*  **cli:** system error logging ([08a4377](https://github.com/sshwy/yaoj-judger/commit/08a4377296503bef81e8c4b73ffe8de0827b5089))


### Features

* multithread logging ([e862c13](https://github.com/sshwy/yaoj-judger/commit/e862c13f2ba19cd40fec1e02cc791a15ee206e60))
*  **interactive:** receives io arguments ([c624e96](https://github.com/sshwy/yaoj-judger/commit/c624e96f95bb729ba095d232b946d6f13608aa0e))
* error ENOENT handling ([291f053](https://github.com/sshwy/yaoj-judger/commit/291f053e396653b95774fcd187b0aa0bee29d706))


### Tests

* refactor test framework ([b22a1f5](https://github.com/sshwy/yaoj-judger/commit/b22a1f563a7eee5278f0254f58b53852c41bed55))
* fix workflow ([5014151](https://github.com/sshwy/yaoj-judger/commit/5014151dd2a7258221554d6164a86510af758952))
* add retry ([5c7e18a](https://github.com/sshwy/yaoj-judger/commit/5c7e18a1c66d440958c881a65dae5743895f6d6d))
* move tle test ([3f6f4fb](https://github.com/sshwy/yaoj-judger/commit/3f6f4fb908aecf8e7b881804e0dafa1e05fc1d42))
* remove some temporary files ([5415973](https://github.com/sshwy/yaoj-judger/commit/54159734fe627f3d8108649a96f3bebf85a6aec0))
* fix workflow ([de26446](https://github.com/sshwy/yaoj-judger/commit/de26446685dd4194522672df6756ba3e32c2f904))


### Bug Fixes

*  **policy:** allow msync ([4e75f73](https://github.com/sshwy/yaoj-judger/commit/4e75f732a88c34e6f17c55f8fe75aaf1ecbdd240))


### Reverts

* remove check_runner_duplicate_before_fork ([1913c51](https://github.com/sshwy/yaoj-judger/commit/1913c51537ff9d4c997bd9c43e3574ca11dce8ad))
*  **judger:** remove traditional ([bb6d37f](https://github.com/sshwy/yaoj-judger/commit/bb6d37f8112d96ecfd81ef988a444313d7663dae))


### Documentation

* update readme ([10af217](https://github.com/sshwy/yaoj-judger/commit/10af217e8739948ae2c72638ae8427072889d712))


### Contributors

<a href="https://github.com/sshwy">sshwy</a>

<details>
<summary>Commit-Lint commands</summary>
<br />

You can trigger Commit-Lint actions by commenting on this PR:
- `@Commit-Lint merge patch` will merge dependabot PR on "patch" versions (X.X.Y - Y change)
- `@Commit-Lint merge minor` will merge dependabot PR on "minor" versions (X.Y.Y - Y change)
- `@Commit-Lint merge major` will merge dependabot PR on "major" versions (Y.Y.Y - Y change)
- `@Commit-Lint merge disable` will desactivate merge dependabot PR
- `@Commit-Lint review` will approve dependabot PR
- `@Commit-Lint stop review` will stop approve dependabot PR



</details>

sshwy committed:
refactor(judger): interactive

sshwy committed:
feat: multithread logging

sshwy committed:
feat(interactive): receives io arguments

sshwy committed:
test: refactor test framework

sshwy committed:
refactor: flush log

sshwy committed:
fix(policy): allow msync

sshwy committed:
test: fix workflow

sshwy committed:
feat(judger)!: general judger stream redirect

BREAKING CHANGE: general judger arguments changed, which now adds three
file parameters (input, output, err) at the beginning

sshwy committed:
refactor(cli): system error logging

sshwy committed:
test: add retry

sshwy committed:
test: move tle test

sshwy committed:
revert: remove check_runner_duplicate_before_fork

It turns out that this is a useless hook

sshwy committed:
test: remove some temporary files

sshwy committed:
revert(judger): remove traditional

It turns out that traditional judger is covered by gereral

sshwy committed:
chore: drop t.in t.out

sshwy committed:
feat: error ENOENT handling

sshwy committed:
test: fix workflow

sshwy committed:
docs: update readme
